runbot_travis2docker/models/runbot_branch.py

```diff
@@ -33,6 +34,29 @@ def _compute_name_weblate(self):
                         dict(match.groupdict(), branch=branch['branch_name']))
             branch.name_weblate = name
 
+    @tools.ormcache('ssh')
+    def _ssh_keyscan(self, ssh):
+        """This function execute the command 'ssh-keysan' to avoid the question
+        when the command git fetch is excecuted.
+        The question is like to:
+            'Are you sure you want to continue connecting (yes/no)?'"""
+        cmd = ['ssh-keyscan', '-p']
+        match = re.search(r'@(?P<host>[^/]+(?=:|/]))?(?::)?(?P<port>[^/]+)?/',
+                          ssh)
+        if not match:
+            return False
+        cmd.extend(sorted(match.groups('22')))
```

Comment:
You are using named sections so you can get the value here with that  name

Reply:
@moylop260 
Sorry for my short explanation before

![image](https://user-images.githubusercontent.com/1387970/30492661-3dec38c4-9a0f-11e7-910f-38ecb2bc0027.png)

The expression regular made for me has a little bug when the port is not present into the url the group returned has bad position 

For that reason the function "sorted" is required for to fix my little trouble

Sorry that was the only way I found :disappointed:

Reply:
What about use `@(?P<host>[^/]+(?=:|/]))?(?::)?(?P<port>\d+)([^/]+)?/'`?
I mean, for the port your get just digits with `\d+`

Reply:
@moylop260 

I try to use this regular expression, but when the group "port" not exists the regular expression fail

![image](https://user-images.githubusercontent.com/1387970/30497431-da93e936-9a20-11e7-9ebd-7b2dc994c4ba.png)

In this URL you can test the regular expresison 
https://regex101.com/r/GA0BJ7/1

Reply:
So, you need change your regex to support both cases.

Something like as:
https://regex101.com/r/qLelXK/3/

Reply:
@moylop260 
That regular expression is awesome.